app/DataTransferObjects/UserData.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\DataTransferObjects;
+
+readonly class UserData
+{
+    public function __construct(
+        public string $pin,
```

Comment:
**[non-blocker]** I see that you’ve used the acronym PIN ("personal identification number"). However, in Estonia this is referred to as a "code" instead of a "number". It’s not a big deal or anything, but I would still suggest renaming it to `pic`.